src/bytes.rs

```diff
@@ -74,7 +74,17 @@ struct Config {
     /// PCRE2_NO_UTF_CHECK
     utf_check: bool,
     /// use pcre2_jit_compile
-    jit: bool,
+    jit: JITChoice,
+}
+
+#[derive(Clone, Debug)]
+enum JITChoice {
+    /// Never do JIT compilation.
+    Never,
+    /// Always do JIT compilation and return an error if it fails.
+    Always,
+    /// Attempt to do JIT compilation but silently fall back to non-JIT.
+    Attempt,
```

Comment:
Probably better name would be `Auto`?

Reply:
Eh, this is an internal thing. It's not public.